src/Polly.Core/ResiliencePropertyKey.cs

```diff
@@ -0,0 +1,30 @@
+namespace Polly;
+
+#pragma warning disable CA1815 // Override equals and operator equals on value types, this type should never be used as key to dictionary
```

Comment:
This would be quite trivial to just implement as there's only the key property which could just forward to the string in most cases?

Reply:
Initially I had this, but then I just opted for the same approach as .NET where they are not overriding these either:
https://source.dot.net/#System.Net.Http/System/Net/Http/HttpRequestOptionsKey.cs 

However, I can add it just for extra safety. It's simple to implement anyway.

Reply:
Done.